kyuubi-common/src/main/scala/org/apache/kyuubi/service/TFrontendService.scala

```diff
@@ -114,6 +114,12 @@ abstract class TFrontendService(name: String)
     checkInitialized()
     val host = (conf.get(FRONTEND_ADVERTISED_HOST), serverHost) match {
       case (Some(advertisedHost), _) => advertisedHost
+      case (None, Some(h)) if (h.equals("0.0.0.0")) =>
```

Comment:
Let's define a utility function in `JavaUtils` instead of writing `h.equals("0.0.0.0")` many times. Meanwhile, IPv6 might be considered in the future.

According to [Wikipedia](https://en.wikipedia.org/wiki/0.0.0.0), the official name of `0.0.0.0` is "This host on this network", defined at [RFC 1122](https://datatracker.ietf.org/doc/html/rfc1122) section 3.2.1.3.

In Linux, [INADDR_ANY](https://www.man7.org/linux/man-pages/man7/ip.7.html) and [IN6ADDR_ANY_INIT](https://www.man7.org/linux/man-pages/man7/ipv6.7.html) are used to represent IPv4 and IPv6 "This host on this network". So we can name the function "isAnyInetAddress"